src/scripts/build_nst_da.py

```diff
@@ -0,0 +1,332 @@
+"""Script that downloads, builds and uploads the Danish part of the NST dataset.
+
+Usage:
+    python build_nst_da.py <destination_dir>
+"""
+
+import datetime as dt
+import logging
+import multiprocessing as mp
+import re
+import shutil
+import tarfile
+import zipfile
+from pathlib import Path
+
+import click
+import numpy as np
+import pandas as pd
+import requests as rq
+from datasets import Audio, Dataset, DatasetDict
+from tqdm.auto import tqdm
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s | %(message)s")
+logger = logging.getLogger(__name__)
+
+
+BASE_URL = "https://www.nb.no/sbfil/talegjenkjenning/16kHz_2020/dk_2020"
+
+
+DATA_URLS = dict(
+    train_metadata=f"{BASE_URL}/ADB_OD_DAN_0565.tar.gz",
+    train_audio=f"{BASE_URL}/lydfiler_16_begge.tar.gz",
+    test_and_errors=f"{BASE_URL}/supplement_dk.tar.xz",
+    metadata_csvs=f"{BASE_URL}/metadata_dk_csv.zip",
+    readme=f"{BASE_URL}/dk-16khz_reorganized_02.pdf",
+)
+
+
+@click.command("Builds and stores the Danish part of the NST dataset.")
+@click.argument("output_dir", type=click.Path())
+def main(output_dir) -> None:
+    for name, url in DATA_URLS.items():
+        filename = name + get_suffix(url)
+        destination_path = Path(output_dir) / filename
+        stream_download(url=url, destination_path=destination_path)
+        uncompress_file(filename=destination_path)
+    reorganise_files(dataset_dir=output_dir)
+    remove_bad_files()
+    dataset = build_huggingface_dataset()
+
+    logger.info(f"Saving the dataset to {output_dir}...")
+    dataset.save_to_disk(
+        str(output_dir),
+        max_shard_size="50MB",
```

Comment:
Rationale? A standard choice?

Reply:
Quite arbitrary, but I just wanted the shard size to be small, as it makes it possible to get the first samples quickly.